### PR TITLE
feat: Sign and Verify VC (JSONLD Signature)

### LINKED ISSUE #24: Sign and Verify VC
User verifiable package to add signature to VC
Verify signed VC

### PR DESCRIPTION
**Pull request by sandrask**
User verifiable package to add signature to VC
Verify signed VC

Closes #24

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

File: pkg/restapi/vc/operation/operations.go [hunk 1]
```diff
@@ -85,30 +110,64 @@ func (c *Operation) createCredentialHandler(rw http.ResponseWriter, req *http.Re
...
 	rw.WriteHeader(http.StatusCreated)
-	c.writeResponse(rw, validCredential)
+	c.writeResponse(rw, signedVC)
+}
+
+func (c *Operation) signCredential(profile *ProfileResponse, vc *verifiable.Credential) (*verifiable.Credential, error) { // nolint:lll
+	signingCtx := &verifiable.LinkedDataProofContext{
+		Creator:       profile.Creator,
+		SignatureType: profile.SignatureType,
+		Suite:         ed25519signature2018.New(),
```
Comment (troyronda) [inline pkg/restapi/vc/operation/operations.go]: todo: vc library dx

File: pkg/restapi/vc/operation/operations.go [hunk 2]
```diff
@@ -85,30 +110,64 @@ func (c *Operation) createCredentialHandler(rw http.ResponseWriter, req *http.Re
...
 	}
 
 	verified := true
 	message := successMsg
 
-	_, _, err = verifiable.NewCredential(body)
+	// Q: should signature suite this be passed in or handled (loaded automatically) by verifiable package
+	// based on signature type in proof
+	// Q: we should have default implementation for key fetcher in verifiable package as well
+	_, _, err = verifiable.NewCredential(body, verifiable.WithEmbeddedSignatureSuites(ed25519signature2018.New()),
```
Comment (troyronda) [inline pkg/restapi/vc/operation/operations.go]: todo: vc DX

eg, 
- one return value. enable a decode pattern for custom structs.
- call this usage parse

File: pkg/restapi/vc/operation/models.go [hunk 3]
```diff
@@ -12,25 +12,31 @@ import (
...
-	ID        string     `json:"id"`
-	URI       string     `json:"uri"`
-	IssueDate *time.Time `json:"issueDate"`
-	DID       string     `json:"did"`
+	Name          string     `json:"name"`
+	DID           string     `json:"did"`
+	URI           string     `json:"uri"`
+	SignatureType string     `json:"signatureType"`
+	Creator       string     `json:"creator"`
+	IssueDate     *time.Time `json:"issueDate"`
```
Comment (troyronda) [inline pkg/restapi/vc/operation/models.go]: IssueDate?

Comment (troyronda) [reply pkg/restapi/vc/operation/models.go]: also is it used as a date or time?

Comment (sandrask) [reply pkg/restapi/vc/operation/models.go]: I will change this to Created in my next PR

**Pull request merged**